src/test/java/org/prebid/server/bidder/loyal/LoyalBidderTest.java

```diff
@@ -0,0 +1,398 @@
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(xNative);
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorForWrongType() throws JsonProcessingException {
+        // given
+        final ObjectNode bidExt = mapper.createObjectNode()
+                .set("prebid", mapper.createArrayNode());
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                        givenBidResponse(bidBuilder -> bidBuilder.ext(bidExt).id("123")));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors())
+                .containsExactly(
+                        BidderError.badServerResponse("bid.ext.prebid.type is not present for bid.id: '123'"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorForAudioType() throws JsonProcessingException {
+        // given
+        final ObjectNode bidExt = mapper.createObjectNode()
+                .putPOJO("prebid", ExtBidPrebid.builder().type(audio).build());
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                        givenBidResponse(bidBuilder -> bidBuilder.ext(bidExt).id("123")));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors())
+                .containsExactly(
+                        BidderError.badServerResponse("Unsupported BidType: audio for bid.id: '123'"));
+    }
+
+    private static BidRequest givenBidRequest(
+            UnaryOperator<BidRequest.BidRequestBuilder> bidRequestCustomizer,
+            UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+
+        return bidRequestCustomizer.apply(BidRequest.builder()
+                        .imp(singletonList(givenImp(impCustomizer))))
+                .build();
+    }
+
+    private static BidRequest givenBidRequest(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return givenBidRequest(UnaryOperator.identity(), impCustomizer);
+    }
+
+    private static Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return impCustomizer.apply(Imp.builder()
+                        .id("123")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpLoyal.of("placementId", "endpointId")))))
+                .build();
+    }
+
+    private static Imp givenImp2(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return impCustomizer.apply(Imp.builder()
+                        .id("321")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpLoyal.of("placementId", "endpointId")))))
+                .build();
+    }
```

Comment:
why do you need two different methods if there is a customizer that can be used for that matter?